core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -86,23 +89,24 @@
 
     private final List<SnapshotShardFailure> shardFailures;
 
-    public SnapshotInfo(String name, List<String> indices, long startTime) {
-        this(name, indices, SnapshotState.IN_PROGRESS, null, Version.CURRENT, startTime, 0L, 0, 0, Collections.emptyList());
+    public SnapshotInfo(String name, String uuid, List<String> indices, long startTime) {
```

Comment:
`SnapshotInfo` is only used within a repository, for example in a GET request: `/_snapshot/my_repo` will return snapshot infos.  Also, `SnapshotInfo` is the structure that is written to the snapshot metadata blob, which again lives in a repository already.  Adding the repository info will also give us another backward compatibility issue to worry about due to the structure being serialized to a blob in the repository.